impl/src/main/java/com/sun/faces/facelets/tag/faces/html/HtmlLibrary.java

```diff
@@ -35,77 +67,77 @@
+        addHtmlComponent("outputLink", HtmlOutputLink.COMPONENT_TYPE, "jakarta.faces.Link");
 
-        addHtmlComponent("outputText", "jakarta.faces.HtmlOutputText", "jakarta.faces.Text");
+        addHtmlComponent("outputText", HtmlOutputText.COMPONENT_TYPE, "jakarta.faces.Text");
 
-        this.addComponent("outputScript", "jakarta.faces.Output", "jakarta.faces.resource.Script", ScriptResourceHandler.class);
+        this.addComponent("outputScript", UIOutput.COMPONENT_TYPE, "jakarta.faces.resource.Script", ScriptResourceHandler.class);
 
-        this.addComponent("outputStylesheet", "jakarta.faces.Output", "jakarta.faces.resource.Stylesheet", StylesheetResourceHandler.class);
+        this.addComponent("outputStylesheet", UIOutput.COMPONENT_TYPE, "jakarta.faces.resource.Stylesheet", StylesheetResourceHandler.class);
 
-        addHtmlComponent("panelGrid", "jakarta.faces.HtmlPanelGrid", "jakarta.faces.Grid");
+        addHtmlComponent("panelGrid", HtmlPanelGrid.COMPONENT_TYPE, "jakarta.faces.Grid");
 
-        addHtmlComponent("panelGroup", "jakarta.faces.HtmlPanelGroup", "jakarta.faces.Group");
+        addHtmlComponent("panelGroup", HtmlPanelGroup.COMPONENT_TYPE, "jakarta.faces.Group");
 
-        addHtmlComponent("selectBooleanCheckbox", "jakarta.faces.HtmlSelectBooleanCheckbox", "jakarta.faces.Checkbox");
+        addHtmlComponent("selectBooleanCheckbox", HtmlSelectBooleanCheckbox.COMPONENT_TYPE, "jakarta.faces.Checkbox");
 
-        addHtmlComponent("selectManyCheckbox", "jakarta.faces.HtmlSelectManyCheckbox", "jakarta.faces.Checkbox");
+        addHtmlComponent("selectManyCheckbox", HtmlSelectManyCheckbox.COMPONENT_TYPE, "jakarta.faces.Checkbox");
 
-        addHtmlComponent("selectManyListbox", "jakarta.faces.HtmlSelectManyListbox", "jakarta.faces.Listbox");
+        addHtmlComponent("selectManyListbox", HtmlSelectManyListbox.COMPONENT_TYPE, "jakarta.faces.Listbox");
 
-        addHtmlComponent("selectManyMenu", "jakarta.faces.HtmlSelectManyMenu", "jakarta.faces.Menu");
+        addHtmlComponent("selectManyMenu", HtmlSelectManyMenu.COMPONENT_TYPE, "jakarta.faces.Menu");
 
-        addHtmlComponent("selectOneListbox", "jakarta.faces.HtmlSelectOneListbox", "jakarta.faces.Listbox");
+        addHtmlComponent("selectOneListbox", HtmlSelectOneListbox.COMPONENT_TYPE, "jakarta.faces.Listbox");
 
-        addHtmlComponent("selectOneMenu", "jakarta.faces.HtmlSelectOneMenu", "jakarta.faces.Menu");
+        addHtmlComponent("selectOneMenu", HtmlSelectOneMenu.COMPONENT_TYPE, "jakarta.faces.Menu");
 
-        addHtmlComponent("selectOneRadio", "jakarta.faces.HtmlSelectOneRadio", "jakarta.faces.Radio");
+        addHtmlComponent("selectOneRadio", HtmlSelectOneRadio.COMPONENT_TYPE, "jakarta.faces.Radio");
 
-        addHtmlComponent("title", "jakarta.faces.Output", "jakarta.faces.Title");
+        addHtmlComponent("title", UIOutput.COMPONENT_TYPE, "jakarta.faces.Title");
     }
```

Comment:
We should probably introduce new constants like `HtmlSelectOneRadio.DEFAULT_RENDERER_TYPE` instead of hardcoding those renderer types as well.

Also I'm wondering where those for "html" and "title" are coming from? They do actually not exist as components ...

Reply:
Especially for the new programmatic facelets, maybe at least the HTML one should exist.